.github/workflows/ci.yml

```diff
@@ -0,0 +1,127 @@
+name: ci
+
+on: [push, pull_request]
+
+env:
+  NODE_VERSION: 10
+  SOLR_VERSION: 6.6.5
+
+jobs:
+
+  build-static:
+    name: Build static files
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          npm-${{ hashFiles('package-lock.json') }}
+          npm-
+    - name: Install dependencies
+      run: npm ci
+    - name: Run webpack build
+      run: npm run build:prod
+    - name: Upload built files
+      uses: actions/upload-artifact@v2
+      with:
+        name: bundles
+        path: bundles/
+    - name: Upload stats file
+      uses: actions/upload-artifact@v2
+      with:
+        name: webpack-stats.json
+        path: webpack-stats.json
+
+  ts-unit:
+    name: TypeScript unit tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup node
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+    - name: Cache node modules
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: npm-${{ hashFiles('package-lock.json') }}
+        restore-keys: |
+          npm-${{ hashFiles('package-lock.json') }}
+          npm-
+    - name: Install dependencies
+      run: npm ci
+    - name: Run Jest
+      run: npx jest --collectCoverage
+    - name: Upload test coverage to Codecov
+      uses: codecov/codecov-action@v1
+      with:
+        flags: typescript
+
+  py-unit:
+    name: Python unit tests
+    runs-on: ubuntu-latest
+    needs: build-static
+    strategy:
+      matrix:
+        python: [3.5, 3.6, 3.7]
+    services:
+      db:
+        image: mariadb
+        env:
+          MYSQL_DATABASE: sandco
+          MYSQL_ROOT_PASSWORD: sandco
+        ports:
+        - 3306:3306
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python }}
+    - name: Cache pip
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ matrix.python }}-${{ hashFiles('requirements.txt') }}
+        restore-keys: |
+          pip-${{ matrix.python }}-${{ hashFiles('requirements.txt') }}
+          pip-${{ matrix.python }}
+          pip-
+    - name: Install dependencies
+      run: |
+        pip install wheel
+        pip install -r requirements.txt
+        pip install -r dev-requirements.txt
+    - name: Install and configure Solr
```

Comment:
Any value in converting this to a reusable github action? (probably for future, not immediate)

I wonder if it would be something we could collaborate on with PUL. (I did some quick searching and didn't find an existing action, but might be worth another look.)

Reply:
potentially. the ideal would be not installing solr on the ci container at all, but instead using [service containers](https://docs.github.com/en/actions/configuring-and-managing-workflows/about-service-containers). i struggled to get this working for quite awhile and eventually abandoned it because I couldn't create the core correctly, but would like to revisit (perhaps with PUL assistance).